drivers/syslog/syslog_channel.c

```diff
@@ -217,16 +217,7 @@ static ssize_t syslog_default_write(FAR struct syslog_channel_s *channel,
                                     FAR const char *buffer, size_t buflen)
 {
 #if defined(CONFIG_ARCH_LOWPUTC)
-  static sem_t sem = SEM_INITIALIZER(1);
-  size_t nwritten;
-
-  nxsem_wait(&sem);
```

Comment:
why the lock is removed? will it not lead to messed up output in multi-tasking syslog usage?

Reply:
We found that calling syslog inside signal handler may happen the dead lock, so it look better to let up_nputs decide how to protect the concurrent access.